frontend/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import React from "react";
+import { apiValidateToken } from "@/lib/Api";
+
+interface AuthContextType {
+    isAuthenticated: boolean;
+    isLoading: boolean;
+}
+
+const AuthContext = React.createContext<AuthContextType>({
+    isAuthenticated: false,
+    isLoading: true,
+});
+
+export const useAuth = () => React.useContext(AuthContext);
+
+export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({
+    children,
+}) => {
+    const [isAuthenticated, setIsAuthenticated] = React.useState(false);
+    const [isLoading, setIsLoading] = React.useState(true);
+
+    React.useEffect(() => {
+        async function fetchAuthStatus() {
+            const res = await apiValidateToken("");
```

Comment:
Was this intentional, to send an empty string as the token?

Reply:
Yep, it's for the middleware.ts to also call the validate token. NextRequest, gets cookie and parses the idToken from it.

Reply:
Gotcha